projects/bp-gallery/src/components/views/exhibitions/ExhibitionViewer.tsx

```diff
@@ -0,0 +1,143 @@
+  const mySection = getSection(sectionId);
+  const [focusedPicture, setFocusedPicture] = useState<string | undefined>(undefined);
+
+  return (
+    <>
+      {focusedPicture && (
+        <Portal container={root}>
+          <PictureView
+            initialPictureId={focusedPicture}
+            siblingIds={mySection?.dragElements
+              .map(drag => drag.picture.id)
+              .filter((value, index, self) => self.indexOf(value) === index)}
+            onBack={() => setFocusedPicture(undefined)}
+          />
+        </Portal>
+      )}
+      {mySection && (
+        <div className='flex flex-col gap-4'>
+          <div className='text-4xl font-semibold'>{mySection.title}</div>
+          <RichText value={mySection.text} />
+          <div className='flex flex-row gap-4 flex-wrap justify-center'>
+            {mySection.dragElements.map((drag, index) => (
+              <div key={index} className='w-max'>
+                <PicturePreview
+                  picture={drag.picture}
+                  height='15rem'
+                  onClick={() => {
+                    setFocusedPicture(drag.picture.id);
+                    pushHistoryWithoutRouter(`/picture/${drag.picture.id}`);
+                  }}
+                />
+              </div>
+            ))}
+          </div>
+        </div>
+      )}
+    </>
+  );
+};
+
+const EndCard = () => {
+  const { t } = useTranslation();
+  const { getEpilog, getSources } = useContext(ExhibitionGetContext);
+  const epilog = getEpilog();
+  const sources = getSources();
+  return (
+    <div className='p-11 box-border gap-4 flex flex-col'>
+      {epilog && (
+        <div className='flex flex-col gap-4'>
+          <div className='text-4xl font-semibold'>{t('exhibition.viewer.epilog')}</div>
+          <RichText value={epilog} />
+        </div>
+      )}
+      {sources && sources.length > 0 && (
+        <div>
+          <div className='text-4xl font-semibold'>{t('exhibition.viewer.sources')}</div>
+          <ul>
+            {sources.map((source, key) => source.source && <li key={key}>{source.source}</li>)}
+          </ul>
+        </div>
+      )}
+    </div>
+  );
+};
+
+const MainPart = () => {
+  const { getAllSections } = useContext(ExhibitionGetContext);
+  const sectionsWithContent = getAllSections()?.filter(
+    section => section.title || section.text || section.dragElements
+  );
+  const sections = sectionsWithContent?.map((section, index) => (
+    <Section key={index} sectionId={section.id} />
+  ));
+  return <div className='p-11 box-border flex flex-col gap-16'>{sections}</div>;
+};
+
+const ExhibitionViewer = ({ exhibitionId }: { exhibitionId: string }) => {
+  const { data: exhibitionData } = useGetExhibitionQuery({
+    variables: { exhibitionId },
+  });
+  const exhibition: FlatExhibition | undefined =
+    useSimplifiedQueryResponseData(exhibitionData)?.exhibition;
+
+  return (
+    <>
+      {exhibition?.is_published && (
```

Comment:
this causes the curators to see a blank page when they try to view their own exhibition before it is published. As a curator, I'd definitely want to see how the exhibition actually looks like before I publish it.